java/compaction/compaction-core/src/main/java/sleeper/compaction/strategy/CompactionStrategy.java

```diff
@@ -25,7 +25,7 @@
 
 public interface CompactionStrategy {
 
-    void init(InstanceProperties instanceProperties, TableProperties tableProperties);
+    void init(InstanceProperties instanceProperties, TableProperties tableProperties, List<FileReference> fileReferences, List<Partition> partitions);
 
-    List<CompactionJob> createCompactionJobs(List<FileReference> activeFilesWithJobId, List<FileReference> activeFilesWithNoJobId, List<Partition> allPartitions);
+    List<CompactionJob> createCompactionJobs();
```

Comment:
Is there any reason to keep init and createCompactionJobs separate if we pass everything into init?